libpod/container.go

```diff
@@ -588,3 +589,36 @@ func (c *Container) Export(path string) error {
 func (c *Container) Commit() (*storage.Image, error) {
 	return nil, ErrNotImplemented
 }
+
+// Wait blocks on a container to exit and returns its exit code
+func (c *Container) Wait() (int32, error) {
+	err := wait.PollImmediateInfinite(1,
+		func() (bool, error) {
+			stopped, err := c.isStopped()
+			if err != nil {
+				return false, err
+			}
+			if !stopped {
+				return false, nil
+			} else { // nolint
+				return true, nil // nolint
+			} // nolint
+		},
+	)
+	if err != nil {
+		return 0, err
+	}
+	exitCode := c.state.ExitCode
+	return exitCode, nil
+}
+
+func (c *Container) isStopped() (bool, error) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	_ = c.syncContainer()
```

Comment:
This is the error we care about - don't need the c.State() call afterwards. syncContainer() will sync with the database, determine if the container was deleted (returning an error if so), and update status from runc (determining if the container stopped). Calling c.State() will do syncContainer() again, but report the error this time (and also it will probably hang because it will try to acquire the locks we grabbed earlier in the function).

Sorry about this - I think I did a terrible job of explaining things initially.